Unable to distribute to testflight builds

### Question Checklist

- [X] Updated fastlane to the latest version
- [X] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [X] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [X] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Question Subject
Error when distributing testflight builds via pilot. 

#### Question Description
We started seeing these errors in our CI pipe line a couple weeks ago that says we are unable to update the change log. But seems like there is something else going on here. 
As we haven't made any changes to our fastlane scripts, anyone else having issues like this starting a couple weeks ago?

Here's are verbose log:

```
ERROR [2019-01-07 17:11:12.56]: fastlane finished with errors
Traceback (most recent call last):
	24: from /Users/chwon/.rvm/gems/ruby-2.5.1/bin/ruby_executable_hooks:24:in `<main>'
	23: from /Users/chwon/.rvm/gems/ruby-2.5.1/bin/ruby_executable_hooks:24:in `eval'
	22: from /Users/chwon/.rvm/gems/ruby-2.5.1/bin/fastlane:23:in `<main>'
	21: from /Users/chwon/.rvm/gems/ruby-2.5.1/bin/fastlane:23:in `load'
	20: from /Users/chwon/.rvm/gems/ruby-2.5.1@global/gems/fastlane-2.113.0/bin/fastlane:23:in `<top (required)>'
	19: from /Users/chwon/.rvm/rubies/ruby-2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.113.0/fastlane/lib/fastlane/cli_tools_distributor.rb:108:in `take_off'
	18: from /Users/chwon/.rvm/rubies/ruby-2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.113.0/fastlane/lib/fastlane/commands_generator.rb:41:in `start'
	17: from /Users/chwon/.rvm/rubies/ruby-2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.113.0/fastlane/lib/fastlane/commands_generator.rb:333:in `run'
	16: from /Users/chwon/.rvm/gems/ruby-2.5.1/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
	15: from /Users/chwon/.rvm/rubies/ruby-2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.113.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:76:in `run!'
	14: from /Users/chwon/.rvm/gems/ruby-2.5.1/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
	13: from /Users/chwon/.rvm/gems/ruby-2.5.1/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
	12: from /Users/chwon/.rvm/gems/ruby-2.5.1/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
	11: from /Users/chwon/.rvm/rubies/ruby-2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.113.0/fastlane/lib/fastlane/commands_generator.rb:108:in `block (2 levels) in run'
	10: from /Users/chwon/.rvm/rubies/ruby-2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.113.0/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
	 9: from /Users/chwon/.rvm/rubies/ruby-2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.113.0/fastlane/lib/fastlane/lane_manager.rb:56:in `cruise_lane'
	 8: from /Users/chwon/.rvm/rubies/ruby-2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.113.0/fastlane/lib/fastlane/runner.rb:45:in `execute'
	 7: from /Users/chwon/.rvm/rubies/ruby-2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.113.0/fastlane/lib/fastlane/runner.rb:45:in `chdir'
	 6: from /Users/chwon/.rvm/rubies/ruby-2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.113.0/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
	 5: from /Users/chwon/.rvm/rubies/ruby-2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.113.0/fastlane/lib/fastlane/lane.rb:33:in `call'
	 4: from ../../Common/fastlane/Fastfile:382:in `block (2 levels) in parsing_binding'
	 3: from /Users/chwon/.rvm/rubies/ruby-2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.113.0/pilot/lib/pilot/build_manager.rb:88:in `distribute'
	 2: from /Users/chwon/.rvm/rubies/ruby-2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.113.0/pilot/lib/pilot/build_manager.rb:92:in `rescue in distribute'
	 1: from /Users/chwon/.rvm/rubies/ruby-2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.113.0/fastlane_core/lib/fastlane_core/ui/ui.rb:17:in `method_missing'
/Users/chwon/.rvm/rubies/ruby-2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.113.0/fastlane_core/lib/fastlane_core/ui/interface.rb:141:in `user_error!': [!] Could not set changelog:


<html> (FastlaneCore::Interface::FastlaneError)
<head>
<meta http-equiv="Content-Type" content="text/html;charset=ISO-8859-1"/>
<title>Error 409 </title>
</head>
<body>
<h2>HTTP ERROR: 409</h2>
<p>Problem accessing /v2/providers/302756/apps/1382945144/builds/37665298. Reason:
<pre>    ERROR-1: Enter a demo account user name, or deselect "A demo account is required to use all the features of the app."; ERROR-1: Enter demo account password, or deselect "A demo account is required to use all the features of the app."</pre></p>
<hr /><i><small>Powered by Jetty://</small></i>
</body>
</html>
```



Other jobs failed with this error: The build is not in state for internal testing. - Build is not in internal testing state.

```
/usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.112.0/spaceship/lib/spaceship/connect_api/client.rb:84:in `handle_response': [!] The build is not in state for internal testing. - Build is not in internal testing state. (Spaceship::UnexpectedResponse)
	from /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.112.0/spaceship/lib/spaceship/connect_api/client.rb:66:in `post_for_testflight_review'
	from /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.112.0/spaceship/lib/spaceship/test_flight/build.rb:223:in `submit_for_testflight_review!'
	from /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.112.0/pilot/lib/pilot/build_manager.rb:183:in `distribute_build'
	from /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.112.0/pilot/lib/pilot/build_manager.rb:108:in `distribute'
	from ../../Common/fastlane/Fastfile:382:in `block (2 levels) in parsing_binding'
	from /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.112.0/fastlane/lib/fastlane/lane.rb:33:in `call'
	from /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.112.0/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
	from /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.112.0/fastlane/lib/fastlane/runner.rb:45:in `chdir'
	from /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.112.0/fastlane/lib/fastlane/runner.rb:45:in `execute'
	from /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.112.0/fastlane/lib/fastlane/lane_manager.rb:56:in `cruise_lane'
	from /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.112.0/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
	from /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.112.0/fastlane/lib/fastlane/commands_generator.rb:108:in `block (2 levels) in run'
	from /usr/local/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
	from /usr/local/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
	from /usr/local/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
	from /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.112.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:76:in `run!'
	from /usr/local/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
	from /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.112.0/fastlane/lib/fastlane/commands_generator.rb:333:in `run'
	from /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.112.0/fastlane/lib/fastlane/commands_generator.rb:41:in `start'
	from /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.112.0/fastlane/lib/fastlane/cli_tools_distributor.rb:108:in `take_off'
	from /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.112.0/bin/fastlane:23:in `<top (required)>'
	from /usr/local/bin/fastlane:23:in `load'
	from /usr/local/bin/fastlane:23:in `<main>'
```

Does not fix the issue for our builds 😔
```
2019-03-05T01:14:00.4929920Z Spaceship::UnexpectedResponse: [31m[!] The build is not in state for internal testing. - Build is not in internal testing state.[0m
2019-03-05T01:14:00.4930960Z   /drop/vendor/bundle/ruby/2.3.0/gems/fastlane-2.117.1/spaceship/lib/spaceship/connect_api/client.rb:274:in `handle_response'
2019-03-05T01:14:00.4931740Z   /drop/vendor/bundle/ruby/2.3.0/gems/fastlane-2.117.1/spaceship/lib/spaceship/connect_api/client.rb:256:in `post_beta_app_review_submissions'
2019-03-05T01:14:00.4932500Z   /drop/vendor/bundle/ruby/2.3.0/gems/fastlane-2.117.1/spaceship/lib/spaceship/test_flight/build.rb:223:in `submit_for_testflight_review!'
2019-03-05T01:14:00.4933170Z   /drop/vendor/bundle/ruby/2.3.0/gems/fastlane-2.117.1/pilot/lib/pilot/build_manager.rb:217:in `distribute_build'
2019-03-05T01:14:00.4933860Z   /drop/vendor/bundle/ruby/2.3.0/gems/fastlane-2.117.1/pilot/lib/pilot/build_manager.rb:93:in `distribute'
2019-03-05T01:14:00.4934530Z   /drop/vendor/bundle/ruby/2.3.0/gems/fastlane-2.117.1/pilot/lib/pilot/build_manager.rb:61:in `upload'
2019-03-05T01:14:00.4935200Z   /drop/vendor/bundle/ruby/2.3.0/gems/fastlane-2.117.1/fastlane/lib/fastlane/actions/upload_to_testflight.rb:16:in `run'
2019-03-05T01:14:00.4936560Z   /drop/vendor/bundle/ruby/2.3.0/gems/fastlane-2.117.1/fastlane/lib/fastlane/runner.rb:253:in `block (2 levels) in execute_action'
2019-03-05T01:14:00.4937240Z   /drop/vendor/bundle/ruby/2.3.0/gems/fastlane-2.117.1/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
2019-03-05T01:14:00.4937920Z   /drop/vendor/bundle/ruby/2.3.0/gems/fastlane-2.117.1/fastlane/lib/fastlane/runner.rb:232:in `block in execute_action'
2019-03-05T01:14:00.4939000Z   /drop/vendor/bundle/ruby/2.3.0/gems/fastlane-2.117.1/fastlane/lib/fastlane/runner.rb:227:in `chdir'
2019-03-05T01:14:00.4939670Z   /drop/vendor/bundle/ruby/2.3.0/gems/fastlane-2.117.1/fastlane/lib/fastlane/runner.rb:227:in `execute_action'
2019-03-05T01:14:00.4940370Z   /drop/vendor/bundle/ruby/2.3.0/gems/fastlane-2.117.1/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
2019-03-05T01:14:00.4940890Z   /drop/vendor/bundle/ruby/2.3.0/gems/fastlane-2.117.1/fastlane/lib/fastlane/fast_file.rb:154:in `method_missing'
2019-03-05T01:14:00.4941210Z   Fastfile:205:in `block (2 levels) in parsing_binding'
2019-03-05T01:14:00.4941610Z   /drop/vendor/bundle/ruby/2.3.0/gems/fastlane-2.117.1/fastlane/lib/fastlane/lane.rb:33:in `call'
2019-03-05T01:14:00.4942870Z   /drop/vendor/bundle/ruby/2.3.0/gems/fastlane-2.117.1/fastlane/lib/fastlane/runner.rb:204:in `try_switch_to_lane'
2019-03-05T01:14:00.4943310Z   /drop/vendor/bundle/ruby/2.3.0/gems/fastlane-2.117.1/fastlane/lib/fastlane/runner.rb:146:in `trigger_action_by_name'
2019-03-05T01:14:00.4943730Z   /drop/vendor/bundle/ruby/2.3.0/gems/fastlane-2.117.1/fastlane/lib/fastlane/fast_file.rb:154:in `method_missing'
2019-03-05T01:14:00.4944060Z   Fastfile:193:in `block (2 levels) in parsing_binding'
2019-03-05T01:14:00.4944460Z   /drop/vendor/bundle/ruby/2.3.0/gems/fastlane-2.117.1/fastlane/lib/fastlane/lane.rb:33:in `call'
2019-03-05T01:14:00.4944880Z   /drop/vendor/bundle/ruby/2.3.0/gems/fastlane-2.117.1/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
2019-03-05T01:14:00.4945300Z   /drop/vendor/bundle/ruby/2.3.0/gems/fastlane-2.117.1/fastlane/lib/fastlane/runner.rb:45:in `chdir'
2019-03-05T01:14:00.4945710Z   /drop/vendor/bundle/ruby/2.3.0/gems/fastlane-2.117.1/fastlane/lib/fastlane/runner.rb:45:in `execute'
2019-03-05T01:14:00.4946130Z   /drop/vendor/bundle/ruby/2.3.0/gems/fastlane-2.117.1/fastlane/lib/fastlane/lane_manager.rb:56:in `cruise_lane'
2019-03-05T01:14:00.4946550Z   /drop/vendor/bundle/ruby/2.3.0/gems/fastlane-2.117.1/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
2019-03-05T01:14:00.4947040Z   /drop/vendor/bundle/ruby/2.3.0/gems/fastlane-2.117.1/fastlane/lib/fastlane/commands_generator.rb:108:in `block (2 levels) in run'
2019-03-05T01:14:00.4947460Z   /drop/vendor/bundle/ruby/2.3.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
2019-03-05T01:14:00.4948210Z   /drop/vendor/bundle/ruby/2.3.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
2019-03-05T01:14:00.4948630Z   /drop/vendor/bundle/ruby/2.3.0/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
2019-03-05T01:14:00.4949060Z   /drop/vendor/bundle/ruby/2.3.0/gems/fastlane-2.117.1/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:76:in `run!'
2019-03-05T01:14:00.4949470Z   /drop/vendor/bundle/ruby/2.3.0/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
2019-03-05T01:14:00.4950140Z   /drop/vendor/bundle/ruby/2.3.0/gems/fastlane-2.117.1/fastlane/lib/fastlane/commands_generator.rb:333:in `run'
2019-03-05T01:14:00.4950560Z   /drop/vendor/bundle/ruby/2.3.0/gems/fastlane-2.117.1/fastlane/lib/fastlane/commands_generator.rb:41:in `start'
2019-03-05T01:14:00.4951000Z   /drop/vendor/bundle/ruby/2.3.0/gems/fastlane-2.117.1/fastlane/lib/fastlane/cli_tools_distributor.rb:119:in `take_off'
2019-03-05T01:14:00.4951400Z   /drop/vendor/bundle/ruby/2.3.0/gems/fastlane-2.117.1/bin/fastlane:23:in `<top (required)>'
2019-03-05T01:14:00.4951770Z   /drop/vendor/bundle/ruby/2.3.0/bin/fastlane:23:in `load'
2019-03-05T01:14:00.4952160Z   /drop/vendor/bundle/ruby/2.3.0/bin/fastlane:23:in `<top (required)>'
2019-03-05T01:14:00.5444320Z ##[error]Bash exited with code '1'.
```

--- a/pilot/lib/pilot/build_manager.rb
+++ b/pilot/lib/pilot/build_manager.rb
@@ -50,6 +50,12 @@ module Pilot
       end
 
       UI.message("If you want to skip waiting for the processing to be finished, use the `skip_waiting_for_build_processing` option")
+      latest_build = wait_for_build_processing_to_be_complete
+      distribute(options, build: latest_build)
+    end
+
+    def wait_for_build_processing_to_be_complete
+      platform = fetch_app_platform
       app_version = FastlaneCore::IpaFileAnalyser.fetch_app_version(config[:ipa])
       app_build = FastlaneCore::IpaFileAnalyser.fetch_app_build(config[:ipa])
       latest_build = FastlaneCore::BuildWatcher.wait_for_build_processing_to_be_complete(app_id: app.apple_id, platform: platform, train_version: app_version, build_version: app_build, poll_interval: config[:wait_processing_interval], strict_build_watch: config[:wait_for_uploaded_build])
@@ -58,7 +64,7 @@ module Pilot
         UI.important("Uploaded app #{app_version} - #{app_build}, but received build #{latest_build.train_version} - #{latest_build.build_version}. If you want to wait for uploaded build to be finished processing, use the `wait_for_uploaded_build` option")
       end
 
-      distribute(options, build: latest_build)
+      return latest_build
     end
 
     def distribute(options, build: nil)
@@ -210,7 +216,7 @@ module Pilot
       UI.message("Distributing new build to testers: #{uploaded_build.train_version} - #{uploaded_build.build_version}")
 
       # This is where we could add a check to see if encryption is required and has been updated
-      check_export_compliance(uploaded_build)
+      set_export_compliance_if_needed(uploaded_build, options)
 
       if options[:groups] || options[:distribute_external]
         begin
@@ -241,32 +247,24 @@ module Pilot
         if external_group.nil? && options[:groups].nil?
           UI.user_error!("You must specify at least one group using the `:groups` option to distribute externally")
         end
-      else # distribute internally
-        # in case any changes to export_compliance are required
-        if uploaded_build.export_compliance_missing?
-          uploaded_build.save!
-        end
       end
 
       true
     end
 
-    def check_export_compliance(uploaded_build)
-      build = uploaded_build.get_app_store_connect_build
-      puts "cec: #{build}"
-      if build["usesNonExemptEncryption"].nil?
-        attributes = {usesNonExemptEncryption: false}
+    def set_export_compliance_if_needed(uploaded_build, options)
+      build = uploaded_build.find_app_store_connect_build
+      build_attributes = build["attributes"] || {}
+      if build_attributes["usesNonExemptEncryption"].nil?
+        uses_non_exempt_encryption = options[:uses_non_exempt_encryption]
+        attributes = { usesNonExemptEncryption: uses_non_exempt_encryption }
 
-        puts "about to send to patch build: #{attributes}"
         client = Spaceship::ConnectAPI::Base.client
-        resp = client.patch_builds(build_id: build["id"], attributes: attributes)
-        puts "resp: #{resp}"
+        client.patch_builds(build_id: build["id"], attributes: attributes)
 
-        UI.message("Waiting for this to process again")
-        platform = fetch_app_platform
-        app_version = FastlaneCore::IpaFileAnalyser.fetch_app_version(config[:ipa])
-        app_build = FastlaneCore::IpaFileAnalyser.fetch_app_build(config[:ipa])
-        latest_build = FastlaneCore::BuildWatcher.wait_for_build_processing_to_be_complete(app_id: app.apple_id, platform: platform, train_version: app_version, build_version: app_build, poll_interval: config[:wait_processing_interval], strict_build_watch: config[:wait_for_uploaded_build])
+        UI.important("Export comlpiance has been set to '#{uses_non_exempt_encryption}'. Need to wait for build to finishing processing again...")
+        UI.important("Set 'ITSAppUsesNonExemptEncryption' in the 'Info.plist' to skip this step and speed up the submission")
+        wait_for_build_processing_to_be_complete
       end
     end
 

--- a/pilot/lib/pilot/build_manager.rb
+++ b/pilot/lib/pilot/build_manager.rb
@@ -210,7 +210,7 @@ module Pilot
       UI.message("Distributing new build to testers: #{uploaded_build.train_version} - #{uploaded_build.build_version}")
 
       # This is where we could add a check to see if encryption is required and has been updated
-      uploaded_build.export_compliance.encryption_updated = false
+      check_export_compliance(uploaded_build)
 
       if options[:groups] || options[:distribute_external]
         begin
@@ -249,6 +249,25 @@ module Pilot
       end
 
       true
+    end
+
+    def check_export_compliance(uploaded_build)
+      build = uploaded_build.get_app_store_connect_build
+      puts "cec: #{build}"
+      if build["usesNonExemptEncryption"].nil?
+        attributes = {usesNonExemptEncryption: false}
+
+        puts "about to send to patch build: #{attributes}"
+        client = Spaceship::ConnectAPI::Base.client
+        resp = client.patch_builds(build_id: build["id"], attributes: attributes)
+        puts "resp: #{resp}"
+
+        UI.message("Waiting for this to process again")
+        platform = fetch_app_platform
+        app_version = FastlaneCore::IpaFileAnalyser.fetch_app_version(config[:ipa])
+        app_build = FastlaneCore::IpaFileAnalyser.fetch_app_build(config[:ipa])
+        latest_build = FastlaneCore::BuildWatcher.wait_for_build_processing_to_be_complete(app_id: app.apple_id, platform: platform, train_version: app_version, build_version: app_build, poll_interval: config[:wait_processing_interval], strict_build_watch: config[:wait_for_uploaded_build])
+      end
     end
 
     def update_review_detail(app_id, info)

--- a/pilot/lib/pilot/options.rb
+++ b/pilot/lib/pilot/options.rb
@@ -134,6 +134,12 @@ module Pilot
                                      default_value: false),
 
         # distribution
+        FastlaneCore::ConfigItem.new(key: :uses_non_exempt_encryption,
+                                     short_option: "-X",
+                                     env_name: "PILOT_USES_NON_EXEMPT_ENCRYPTION",
+                                     description: "Provide the 'Uses Non-Exempt Encryption' for export compliance. This is used if there is 'ITSAppUsesNonExemptEncryption' is not set in the Info.plist",
+                                     default_value: false,
+                                     type: Boolean),
         FastlaneCore::ConfigItem.new(key: :distribute_external,
                                      is_string: false,
                                      env_name: "PILOT_DISTRIBUTE_EXTERNAL",

--- a/pilot/spec/build_manager_spec.rb
+++ b/pilot/spec/build_manager_spec.rb
@@ -77,6 +77,9 @@ describe "Build Manager" do
       })
     end
     let(:mock_api_client_builds) do
+      [{ "id" => "123", "attributes" => { "usesNonExemptEncryption" => true } }]
+    end
+    let(:mock_api_client_builds_with_nil_encryption) do
       [{ "id" => "123" }]
     end
     let(:mock_api_client_build_beta_details) do
@@ -147,7 +150,8 @@ describe "Build Manager" do
           demo_account_required: true,
           notify_external_testers: true,
           beta_app_feedback_email: "josh+oldfeedback@rokkincat.com",
-          beta_app_description: "old description for all the things"
+          beta_app_description: "old description for all the things",
+          uses_non_exempt_encryption: false
         }
       end
 
@@ -166,10 +170,14 @@ describe "Build Manager" do
 
       it "updates non-localized  demo_account_required, notify_external_testers, beta_app_feedback_email, and beta_app_description" do
         options = distribute_options_non_localized
+        builds = mock_api_client_builds_with_nil_encryption
 
+        # Receive 1: finding build for patching review information
+        # Receive 2: finding build for patching uses non-exempt encryption
+        # Receive 3: finding build for submitting for review
         expect(mock_base_api_client).to receive(:get_builds).with({
           filter: { expired: false, processingState: "PROCESSING,VALID", version: ready_to_submit_mock_build.build_version }
-        }).and_return(mock_api_client_builds)
+        }).and_return(builds).exactly(3).times
 
         # Demo account
         expect(mock_base_api_client).to receive(:patch_beta_app_review_detail).with({
@@ -179,7 +187,7 @@ describe "Build Manager" do
 
         # Auto notify
         expect(mock_base_api_client).to receive(:get_build_beta_details).with({
-          filter: { build: mock_api_client_builds.first['id'] }
+          filter: { build: builds.first['id'] }
         }).and_return(mock_api_client_build_beta_details)
         expect(mock_base_api_client).to receive(:patch_build_beta_details).with({
           build_beta_details_id: mock_api_client_build_beta_details.first['id'],
@@ -201,7 +209,12 @@ describe "Build Manager" do
         end
         expect(FastlaneCore::UI).to receive(:success).with("Successfully set the beta_app_feedback_email and/or beta_app_description")
 
-        expect(fake_build_manager).to receive(:distribute_build)
+        expect(FastlaneCore::UI).to receive(:message).with(/Distributing new build to testers/)
+        expect(mock_base_api_client).to receive(:patch_builds).with({
+          build_id: builds.first["id"], attributes: { usesNonExemptEncryption: false }
+        }).and_return(mock_api_client_beta_app_localizations)
+        expect(fake_build_manager).to receive(:wait_for_build_processing_to_be_complete)
+
         expect(FastlaneCore::UI).to receive(:success).with(/Successfully distributed build to/)
 
         fake_build_manager.distribute(options, build: ready_to_submit_mock_build)

--- a/spaceship/lib/spaceship/connect_api/client.rb
+++ b/spaceship/lib/spaceship/connect_api/client.rb
@@ -230,6 +230,27 @@ module Spaceship
         handle_response(response)
       end
 
+      def patch_builds(build_id: nil, attributes: {})
+        # PATCH
+        # https://appstoreconnect.apple.com/iris/v1/builds/<build_id>
+        path = "builds/#{build_id}"
+
+        body = {
+          data: {
+            attributes: attributes,
+            id: build_id,
+            type: "builds"
+          }
+        }
+
+        response = request(:patch) do |req|
+          req.url(path)
+          req.body = body.to_json
+          req.headers['Content-Type'] = 'application/json'
+        end
+        handle_response(response)
+      end
+
       def post_beta_app_review_submissions(build_id: nil)
         # POST
         # https://appstoreconnect.apple.com/iris/v1/betaAppReviewSubmissions

--- a/spaceship/lib/spaceship/test_flight/build.rb
+++ b/spaceship/lib/spaceship/test_flight/build.rb
@@ -217,9 +217,7 @@ module Spaceship
         return if ready_to_test?
         return if approved?
 
-        resp = Spaceship::ConnectAPI::Base.client.get_builds(filter: { expired: false, processingState: "PROCESSING,VALID", version: self.build_version })
-        build = resp.first
-
+        build = get_app_store_connect_build
         Spaceship::ConnectAPI::Base.client.post_beta_app_review_submissions(build_id: build["id"])
       end
 
@@ -229,6 +227,12 @@ module Spaceship
 
       def add_group!(group)
         client.add_group_to_build(app_id: app_id, group_id: group.id, build_id: id)
+      end
+
+      # Bridges the TestFlight::Build to the App Store Connect API build
+      def get_app_store_connect_build
+        resp = Spaceship::ConnectAPI::Base.client.get_builds(filter: { expired: false, processingState: "PROCESSING,VALID", version: self.build_version })
+        resp.first
       end
     end
   end

--- a/spaceship/lib/spaceship/test_flight/build.rb
+++ b/spaceship/lib/spaceship/test_flight/build.rb
@@ -217,7 +217,7 @@ module Spaceship
         return if ready_to_test?
         return if approved?
 
-        build = get_app_store_connect_build
+        build = find_app_store_connect_build
         Spaceship::ConnectAPI::Base.client.post_beta_app_review_submissions(build_id: build["id"])
       end
 
@@ -230,7 +230,7 @@ module Spaceship
       end
 
       # Bridges the TestFlight::Build to the App Store Connect API build
-      def get_app_store_connect_build
+      def find_app_store_connect_build
         resp = Spaceship::ConnectAPI::Base.client.get_builds(filter: { expired: false, processingState: "PROCESSING,VALID", version: self.build_version })
         resp.first
       end

